src/NzbDrone.Core/Download/Clients/Deluge/DelugeProxy.cs

```
@@ -157,134 +159,134 @@ public void SetLabel(string hash, string label, DelugeSettings settings)
+                        return response.Result;
                     }
+                    error = response.Error.ToObject<DelugeError>();
 
-                    throw new DownloadClientAuthenticationException(response.Error.Message);
+                    throw new DownloadClientAuthenticationException(error.Message);
                 }
 
-                throw new DelugeException(response.Error.Message, response.Error.Code);
+                throw new DelugeException(error.Message, error.Code);
             }
 
-            return response;
+            return response.Result;
         }
-        
-        private DelugeResponse<TResult> ProcessRequest<TResult>(IRestClient client, string action, object[] arguments)
+
+        private JsonRpcResponse<TResult> ProcessRequest<TResult>(JsonRpcRequestBuilder requestBuilder, string method, params object[] arguments)
         {
-            var request = new RestRequest(Method.POST);
-            request.Resource = "json";
-            request.RequestFormat = DataFormat.Json;
-            request.AddHeader("Accept-Encoding", "gzip,deflate");
+            var request = requestBuilder.Call(method, arguments).Build();
 
-            var data = new Dictionary<string, object>();
-            data.Add("id", GetCallId());
-            data.Add("method", action);
+            HttpResponse response;
+            try
+            {
+                response = _httpClient.Execute(request);
 
-            if (arguments != null)
+                return Json.Deserialize<JsonRpcResponse<TResult>>(response.Content);
+            }
+            catch (HttpException ex)
             {
-                data.Add("params", arguments);
+                if (ex.Response.StatusCode == HttpStatusCode.RequestTimeout)
+                {
+                    _logger.Debug("Deluge timeout during request, daemon connection may have been broken. Attempting to reconnect.");
+                    return new JsonRpcResponse<TResult>()
```

Should we throw an exception here then reconnect?